mantidproject/mantid
# Add scaling for detectors to SaveLauenorm

Description of work.
SaveLaueNorm has a checkbox option to use the same instrument parameters that are used in SaveHKL for scaling detector banks. They can be set with SetDetScale algorithm for testing. This is a request from MANDI instrument scientist.  The unit test was modified to check that this works.
**To test:**
Check that the intensity and sigI are half after DetScale=0.5:
https://www.dropbox.com/s/kz3udbt2vbuouih/new.integrate?dl=0



``` python
import os

#load a peaks workspace from file
peaks = LoadIsawPeaks(Filename=r'new.integrate')
SetDetScale(Workspace='peaks', DetScaleList='1:0.5,2:1.5')
print "Number of peaks in table %d" % peaks.rowCount()

prefix = os.path.expanduser("~/MyPeaks")
for bool in [False, True]:
    SaveLauenorm(InputWorkspace=peaks, Filename=prefix, MinWavelength=1., MaxWavelength=2.,MinDSpacing=0.5, SortFilesBy='Bank',UseDetScale=bool)
    print "UseDetScale is set to ", bool
    finalfile = prefix + "001"
    ifile = open(finalfile, 'r')
    lines = ifile.readlines()
    for line in lines:
        print line
    ifile.close()
    print "Number of peaks in table %d" % len(lines)
```

Fixes #17498



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
